pkg/quota/controller/clusterquotamapping/lockfactory.go

```diff
@@ -0,0 +1,40 @@
+package clusterquotamapping
+
+import (
+	"sync"
+)
+
+type LockFactory interface {
+	GetLock(string) sync.Locker
+}
+
+type DefaultLockFactory struct {
+	lock sync.RWMutex
+
+	locks map[string]sync.Locker
+}
+
+func NewDefaultLockFactory() *DefaultLockFactory {
```

Comment:
Same thing for the NewSynchronizedMap. Are you planning on using them in a follow-up to this?